### PR TITLE
consolidating OnlineDDL 'singleton' tests into 'scheduler' tests: part 1

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Following up on the discussion on https://github.com/vitessio/vitess/pull/11988#discussion_r1060209136

In this PR we copy all `onlineddl_singleton` tests into `onlineddl_scheduler`. A bit of refactoring was required to adapt to the subtle differences between the two.

The objective is to reduce the number of CI workflows.

Once this PR is merged, we will follow up in a next PR that completely removes `onlineddl_singleton` from the code base.

## Related Issue(s)

#6926

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/onlineddl/scheduler/onlineddl_scheduler_test.go [hunk 1]
```diff
@@ -671,82 +764,318 @@ func TestSchemaChange(t *testing.T) {
...
+		DROP TABLE stress_test
+	`
+		dropNonexistentTableStatement = `
+		DROP TABLE IF EXISTS t_non_existent
+	`
+		multiDropStatements = `DROP TABLE IF EXISTS t1; DROP TABLE IF EXISTS t2; DROP TABLE IF EXISTS t3;`
+	)
+
+	// func testOnlineDDLStatement(t *testing.T, ddlStatement string, ddlStrategy string, executeStrategy string,                            expectHint string, expectError string, skipWait bool) (uuid string) {
+	// func testOnlineDDLStatement(t *testing.T, alterStatement string, ddlStrategy string, executeStrategy string, migrationContext string, expectHint string, expectError string, skipWait bool) (uuid string) {
```
Comment (mattlord) [inline go/test/endtoend/onlineddl/scheduler/onlineddl_scheduler_test.go:834]: Nit, but we should remove these I think.

**Review by mattlord — APPROVED** [review]
Thanks! ❤️

File: go/test/endtoend/onlineddl/scheduler/onlineddl_scheduler_test.go (hunk 1 above)
Comment (shlomi-noach) [reply go/test/endtoend/onlineddl/scheduler/onlineddl_scheduler_test.go:834]: whoops. removed.

**Review by deepthi — APPROVED** [review]
I would have preferred the remove of `singleton` tests to happen in the same PR, but this is ok for now.

**Comment by shlomi-noach** [comment]
> I would have preferred the removal of singleton tests to happen in the same PR, but this is ok for now.

@deepthi the problem I see with that is a limbo between the `singularity` endtoend file existing/not existing, and `project settings->protected branches->required tests` existing/not existing. Some open branches may get confused by this?

**Comment by shlomi-noach** [comment]
similar: https://github.com/vitessio/vitess/pull/12061

**Comment by shlomi-noach** [comment]
Followup in https://github.com/vitessio/vitess/pull/12182

**Pull request merged**